src/visitor.rs

```diff
@@ -132,22 +133,44 @@ impl<'a> FmtVisitor<'a> {
         self.buffer.push_str("{");
 
         if self.config.remove_blank_lines_at_start_or_end_of_block() {
-            if let Some(stmt) = b.stmts.first() {
-                let snippet = self.snippet(mk_sp(self.last_pos, stmt.span.lo));
-                let len = CommentCodeSlices::new(&snippet)
-                    .nth(0)
-                    .and_then(|(kind, _, s)| {
-                        if kind == CodeCharKind::Normal {
-                            // There may be inner attributes
-                            let s = &s[..s.len() -
-                                           s.trim_left_matches(&[' ', '\t', '\r', '\n'][..]).len()];
-                            s.rfind('\n')
+            if let Some(first_stmt) = b.stmts.first() {
+                let attr_lo = inner_attrs
+                    .and_then(|attrs| {
+                        attrs
+                            .iter()
+                            .filter(|a| a.style == ast::AttrStyle::Inner)
+                            .nth(0)
+                            .map(|attr| attr.span.lo)
+                    })
+                    .or_else(|| {
+                        // syntax BUG: `attrs()` returns nothing for items
```

Comment:
I tried, but test failed within `tests/target/attrib-extern-crate.rs` in the same way as `first_stmt.attrs()`.

https://github.com/rust-lang-nursery/rustfmt/blob/0ee76bec0f6a76b7b26eebdbd5700e9c727a96b3/src/lib.rs#L112-L118 looks like it does nothing to `StmtKind::Item`. Should this be modified to include attributes for items?